common-rest/src/main/java/uk/ac/ebi/uniprot/api/rest/controller/BasicSearchController.java

```diff
@@ -113,4 +126,18 @@ protected BasicSearchController(ApplicationEventPublisher eventPublisher,
 
     protected abstract String getEntityId(T entity);
 
+    protected abstract Optional<String> getEntityRedirectId(T entity);
+
+
+    private String getLocationURLForId(String redirectId) {
+        String path = ServletUriComponentsBuilder
+                .fromCurrentRequest()
+                .build()
+                .getPath();
+        if (path != null) {
+            path = path.substring(0, path.lastIndexOf("/") + 1) + redirectId;
```

Comment:
sonarqube intellij plugin that using single quotes `'/'` in lastIndexOf will be faster than `"/"`.

Reply:
Code changed to char. Good Catch..